functorch/functorch/csrc/BatchRulesBinaryOps.cpp

```diff
@@ -268,6 +268,25 @@ std::tuple<Tensor,optional<int64_t>> cdist_backward_batch_rule(
   return std::make_tuple(out, out_bdim);
 }
 
+void fill__Tensor_batch_rule(
+    Tensor& self,
+    optional<int64_t> self_bdim,
+    const Tensor& other,
+    optional<int64_t> other_bdim) {
+  if (!other_bdim.has_value()) {
+    // Optimization: fill_ is faster than the other path which does
+    // reshaping + copy_
+    self.fill_(other);
+    return;
+  }
+  if (!self_bdim && other_bdim) {
+    vmapIncompatibleInplaceError("fill_");
+  }
+  auto self_and_other = _binary_pointwise_helper(
+      self, self_bdim, other, other_bdim, /*do_type_promotion*/false);
+  std::get<0>(self_and_other).copy_(std::get<1>(self_and_other));
```

Comment:
Why do we use copy_ here instead of fill_?

Reply:
`a.fill_(b)`, where `a, b` are both Tensors, only works if b is a scalar Tensor.

The scalar Tensor case is handled above (if b has no bdim, then it must be a scalar Tensor, so we can call fill_).

Now, for the non-scalar Tensor case, we need to call something that is not `fill_`, and `copy_` gives us the same semantics.